### PR TITLE
in_dummy: optimize message generation.

### PR DESCRIPTION
**Pull request by pwhelan**
Improve the generation of messages by in_dummy by only creating the msgpack once at initialization.

----

**Testing**
- [x] Example configuration file for the change
- [x] Debug log output from testing the change
- [x] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

**Documentation**
- [ ] Documentation required for this feature

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

**Comment by pwhelan** [comment]
Testing the change is as simple as invoking fluent-bit with the dummy input plugin:

`
$ fluent-bit -i dummy -o stdout -p match='*'
`

**Comment by pwhelan** [comment]
Here is the debug log:
[debug.log](https://github.com/fluent/fluent-bit/files/7091965/debug.log)

Here is the valgrind.log:
[valgrind.log](https://github.com/fluent/fluent-bit/files/7091976/valgrind.log)

File: plugins/in_dummy/in_dummy.c [hunk 1]
```diff
@@ -58,27 +58,40 @@ static int set_dummy_timestamp(msgpack_packer *mp_pck, struct flb_dummy *ctx)
...
+    flb_input_chunk_append_raw(ins, NULL, 0, ctx->mp_sbuf.data,
+                               ctx->mp_sbuf.size);
+
+    if (ctx->samples > 0) {
+        ctx->samples_count++;
+    }
+    return 0;
+}
+
+/* cb_collect callback */
```
Comment (nokute78) [inline plugins/in_dummy/in_dummy.c:77]: Could you remove this comment ?

**Review by nokute78 — CHANGES_REQUESTED** [review]
This patch will break the feature `Start_time_sec`/`Start_time_nsec`.
https://docs.fluentbit.io/manual/pipeline/inputs/dummy#configuration-parameters

Could you check below command?
`fluent-bit -i dummy -p Start_time_sec=12345678 -o stdout`

It may be better to create message once when `Start_time_sec` and `Start_time_nsec` are not set.

**Comment by pwhelan** [comment]
> fluent-bit -i dummy -p Start_time_sec=12345678 -o stdout

Here is the output of that command with the latest fixes:

```
./bin/fluent-bit -i dummy -p Start_time_sec=12345678 -o stdout
Fluent Bit v1.9.0
* Copyright (C) 2019-2021 The Fluent Bit Authors
* Copyright (C) 2015-2018 Treasure Data
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2021/09/07 11:38:38] [ info] [engine] started (pid=1156675)
[2021/09/07 11:38:38] [ info] [storage] version=1.1.1, initializing...
[2021/09/07 11:38:38] [ info] [storage] in-memory
[2021/09/07 11:38:38] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2021/09/07 11:38:38] [ info] [cmetrics] version=0.2.1
[2021/09/07 11:38:38] [ info] [sp] stream processor started
[0] dummy.0: [12345678.000000000, {"message"=>"dummy"}]
[1] dummy.0: [12345678.999987143, {"message"=>"dummy"}]
[2] dummy.0: [12345679.999979723, {"message"=>"dummy"}]
[3] dummy.0: [12345680.999982798, {"message"=>"dummy"}]
[4] dummy.0: [12345681.999982361, {"message"=>"dummy"}]
[0] dummy.0: [12345683.000000746, {"message"=>"dummy"}]
[1] dummy.0: [12345683.999985652, {"message"=>"dummy"}]
[2] dummy.0: [12345684.999992767, {"message"=>"dummy"}]
[3] dummy.0: [12345685.999986100, {"message"=>"dummy"}]
[0] dummy.0: [12345687.000015958, {"message"=>"dummy"}]
[1] dummy.0: [12345687.999980789, {"message"=>"dummy"}]
[2] dummy.0: [12345688.999980110, {"message"=>"dummy"}]
[3] dummy.0: [12345689.999984146, {"message"=>"dummy"}]
[4] dummy.0: [12345690.999984761, {"message"=>"dummy"}]
^C[2021/09/07 11:38:53] [engine] caught signal (SIGINT)
[0] dummy.0: [12345692.000015529, {"message"=>"dummy"}]
[2021/09/07 11:38:53] [ warn] [engine] service will stop in 5 seconds
[2021/09/07 11:38:57] [ info] [engine] service stopped
```

With the latest changes the option to pregenerate the message once is called fixed_timestamp (a boolean). This option basically disables any and all of the timestamp features, which makes testing ordering and delivery reliability hard or impossible but makes high performance tests possible. It might be a good idea to add warnings if conflicting options are enabled by the user, ie: start_time_sec/nsec and fixed_timestamp.

**Review by nokute78 — APPROVED** [review]
Thank you. LGTM.

**Pull request merged**